### PR TITLE
Compose message from conversation list

### PR DESCRIPTION
**Pull request by shichongrui**
refs: MBL-12997
affects: parent
release note: none

test plan:
 - Compose a message from the conversation list (make sure the
   experimental feature is on)
 - user in the ticket

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018088&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Parent

[MBL-12997](https://instructure.atlassian.net/browse/MBL-12997)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.45% | 93.42% | 0.03%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Use Cases/GetConversationCourses.swift [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    public var fetchedCourses: [APICourse]?
+    public func makeRequest(environment: AppEnvironment, completionHandler: @escaping ([APIEnrollment]?, URLResponse?, Error?) -> Void) {
+        environment.api.exhaust(request) { (apiEnrollments, response, error) in
+            guard let enrollments = apiEnrollments, error == nil else {
+                completionHandler(apiEnrollments, response, error)
+                return
+            }
+
+            let coursesRequestable = GetCoursesRequest(enrollmentState: .active, state: nil, perPage: 99)
+            environment.api.exhaust(coursesRequestable) { [weak self]  (apiCourses, response, error) in
```
[the remark this answers is not in the document]
Comment (shichongrui) [reply Core/Core/Use Cases/GetConversationCourses.swift]: Oh good eye. Did even see that.

File: Core/Core/Models/Enrollment.swift [hunk 2]
```diff
@@ -156,5 +157,10 @@ extension Enrollment {
                 grades.insert(currentPeriodGrade)
             }
         }
+
+        if let apiObservedUser = item.observed_user {
+            let observedUserModel = User.save(apiObservedUser, in: client)
+            observedUser = observedUserModel
+        }
```
[the remark this answers is not in the document]
Comment (shichongrui) [reply Core/Core/Models/Enrollment.swift]: Yeah I think that makes sense.

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: Parent/Parent/Conversations/ConversationCoursesActionSheet.swift [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        return vc
+    }
+
+    override func viewDidLoad() {
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.separatorStyle = .none
+        tableView.registerCell(SubtitleTableViewCell.self)
+
+        let header = UILabel(frame: CGRect(x: 16, y: 0, width: tableView.frame.width - 32, height: 40))
```
Comment (armstrongnate) [inline Parent/Parent/Conversations/ConversationCoursesActionSheet.swift]: This header looks a hair too small. In figma it's 40 high but has a top padding of 8. if you make the y: 8 will that work? I wonder if the y will get ignored...

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018096&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018097&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Pull request merged**